src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.UniversalDao;
+import core.basesyntax.factory.Factory;
+import core.basesyntax.model.Bet;
+import core.basesyntax.model.User;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoDaoAnnotationInTheClass {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredField = clazz.getDeclaredFields();
+        for (Field field : declaredField) {
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getType().equals(UniversalDao.class)) {
+                field.setAccessible(true);
+                Type type = field.getGenericType();
+                if (type instanceof ParameterizedType) {
+                    ParameterizedType parameterizedType = (ParameterizedType) type;
+                    if (Bet.class == parameterizedType.getActualTypeArguments()[0]) {
+                        field.set(instance, Factory.getBetDao());
+                    }
+                    if (User.class == parameterizedType.getActualTypeArguments()[0]) {
+                        field.set(instance, Factory.getUserDao());
+                    }
+                }
+            }
```

Comment:
looks very interesting that you have found the way to get generic type!

I would still recommend that we do not use generics here

in this code part:

let's have first `if` checking if field has Inject annotation

then inside that `if`:

1. check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2. else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3. else - throw your custom Exception